### PR TITLE
Update Quiver Log for null safety and clean up all warnings.

### PR DESCRIPTION
**Pull request by adambender**
Forgive the mess of a commit history. Its been a long time since I have used git/github :)

### CODE REVIEW

File: lib/src/web.dart [hunk 1]
```diff
@@ -42,6 +42,6 @@ class WebAppender extends Appender {
 
   @override
   void append(LogRecord record, Formatter formatter) {
-    _levelToOutputFunction[record.level](formatter.call(record));
+    _levelToOutputFunction![record.level]!(formatter.call(record));
```
Comment (cbracken) [inline lib/src/web.dart:45]: Similarly, why is the first `!` required here?

File: lib/src/web.dart [hunk 2]
```diff
@@ -15,7 +15,7 @@ part of quiver.log.web;
 ///  Level.SHOUT => console.error
 class WebAppender extends Appender {
   final Console _console;
-  UnmodifiableMapView<Level, Function> _levelToOutputFunction;
+  UnmodifiableMapView<Level, Function>? _levelToOutputFunction;
```
Comment (cbracken) [inline lib/src/web.dart:18]: Why is this nullable? It looks like it's set in the constructor.

File: test/appender_test.dart [hunk 3]
```diff
@@ -71,19 +72,25 @@ class ExceptionalAppender extends Appender {
...
-  StreamController<LogRecord> _controller = StreamController(sync: true);
+  final _controller = StreamController<LogRecord>(sync: true);
+
+  @override
   Stream<LogRecord> get onRecord => _controller.stream;
 
-  void info(message, [Object error, StackTrace stackTrace]) =>
-      _controller.add(LogRecord(Level.INFO, message, 'simple'));
+  @override
+  void info(Object? msg, [Object? message, StackTrace? stackTrace]) =>
```
Comment (cbracken) [inline test/appender_test.dart:81]: Could we make the message non-nullable to avoid logging `null`?

File: test/webappender_test.dart [hunk 4]
```diff
@@ -19,82 +19,82 @@ import 'dart:html';
...
+import 'package:test/test.dart';
 
-main() {
-  WebAppender webAppender;
-  Logger logger;
-  FakeConsole fakeConsole;
+void main() {
+  WebAppender? webAppender;
+  Logger? logger;
+  FakeConsole? fakeConsole;
```
Comment (cbracken) [inline test/webappender_test.dart:27]: All three of these could be `late final` which would avoid all the `!` below.

**Review by cbracken — APPROVED** [review]
Just a few suggestions othewise lgtm.

File: test/webappender_test.dart (hunk 4 above)
Comment (adambender) [reply test/webappender_test.dart]: Done.

File: test/appender_test.dart (hunk 3 above)
Comment (adambender) [reply test/appender_test.dart:81]: This signature comes from the base Logger class.

File: lib/src/web.dart (hunk 2 above)
Comment (adambender) [reply lib/src/web.dart]: Moved to initializer.

File: lib/src/web.dart (hunk 1 above)
Comment (adambender) [reply lib/src/web.dart]: Used late.

**Pull request merged**